### PR TITLE
Move Faraday_Proxy_Tools/FaradayIO to faraday/proxyio

### PR DESCRIPTION
**Pull request by reillyeon**
This patch creates a top-level faraday module which begins with the proxy I/O library. Scripts are updated to import via the new module path. Scripts which are expected to be executed from the command line have a Python #! line added and are marked executable.

### CODE REVIEW

**Comment by kb1lqd** [comment]
@reillyeon I'd like to take a stab at reviewing this unless you really want @kb1lqc to review. It looks as if this creates FaradayIO into a full mini module? 

It looks like the top level servers needed the file paths appended in sys:
* https://github.com/FaradayRF/Faraday-Software/pull/150/commits/096d70e812f498b3d3c2f8c6c4fa44a33c9391f7#diff-2b9506bb756db60f8e6a5123340a651b
* https://github.com/FaradayRF/Faraday-Software/pull/150/commits/096d70e812f498b3d3c2f8c6c4fa44a33c9391f7#diff-70266ca05a76688f6b01443948ca806a

but not the small sub-programs?
* https://github.com/FaradayRF/Faraday-Software/pull/150/commits/096d70e812f498b3d3c2f8c6c4fa44a33c9391f7#diff-6cc78c51651c07935c5337782f4dc97b

**Comment by reillyeon** [comment]
The path only needs to be configured once (and once I get the installation scripts working and documented, not at all) so I've tried to limit it to the runnable scripts so that the modules don't mess with the path.

**Review by kb1lqd — COMMENTED** [review]
Confirmed proxy boots with multiuple units.

File: Applications/Telemetry/telemetry.py [hunk 1]
```diff
@@ -24,10 +25,11 @@
 from flask import request
 from flask_cors import CORS
 
-# Can we clean this up?
-sys.path.append(os.path.join(os.path.dirname(__file__), "../../Faraday_Proxy_Tools/"))  #Append path to common tutorial FaradayIO module
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import telemetryparser
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
```
Comment (kb1lqd) [inline Applications/Telemetry/telemetry.py:29]: I get an import error when trying to run telemetry.py

image

File: Applications/Hermes/hermesflaskserver.py [hunk 2]
```diff
@@ -8,6 +9,11 @@
 from flask import Flask
 from flask import request
 
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
```
Comment (kb1lqd) [inline Applications/Hermes/hermesflaskserver.py:13]: I get another import error when running this script:

image

File: Applications/Telemetry/telemetry.py (hunk 1 above)
Comment (kb1lqd) [reply Applications/Telemetry/telemetry.py:29]: I modified the import paths and got it to work but I may be removing the intended syntax you wanted @reillyeon with the `faraday.proxyio`:

image

**Review by kb1lqd — DISMISSED** [review]
Please comment/address my noted issues.

File: Applications/Telemetry/telemetry.py (hunk 1 above)
Comment (kb1lqd) [reply Applications/Telemetry/telemetry.py:29]: Confirmed fixed with __init__.py file added in commit.

File: Applications/Hermes/hermesflaskserver.py (hunk 2 above)
Comment (kb1lqd) [reply Applications/Hermes/hermesflaskserver.py:13]: Confirmed fixed with __init__.py file added in commit.

**Comment by kb1lqd** [comment]
I confirmed that I could operated successfully simultaneously:

* Proxy
* Telemetry
* APRS
* Hermes Message

File: Applications/Hermes/hermesmessage.py [hunk 3]
```diff
@@ -1,14 +1,8 @@
...
-
-sys.path.append(os.path.join(os.path.dirname(__file__),
-                             "../../Faraday_Proxy_Tools"))  # Append path to common tutorial FaradayIO module
-# noinspection PyPep8
-from FaradayIO import faradaybasicproxyio
-# noinspection PyPep8
-from FaradayIO import faradaycommands
+
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
```
Comment (kb1lqd) [inline Applications/Hermes/hermesmessage.py:5]: Confirmed working.

File: Applications/deviceconfiguration/deviceconfiguration.py [hunk 4]
```diff
@@ -9,13 +11,12 @@
...
-                             "../../Faraday_Proxy_Tools"))  # Append path to common tutorial FaradayIO module
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
 
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import deviceconfig
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import deviceconfig
```
Comment (kb1lqd) [inline Applications/deviceconfiguration/deviceconfiguration.py:19]: Confirmed working with unit test.

File: Applications/deviceconfiguration/readconfig.py [hunk 5]
```diff
@@ -14,13 +16,13 @@
...
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
 
 # Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import telemetryparser
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import telemetryparser
```
Comment (kb1lqd) [inline Applications/deviceconfiguration/readconfig.py:25]: Confirmed working with unit test.

File: Applications/deviceconfiguration/simpleconfig.py [hunk 6]
```diff
@@ -10,12 +12,13 @@
...
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
 
 #Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import telemetryparser
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import telemetryparser
```
Comment (kb1lqd) [inline Applications/deviceconfiguration/simpleconfig.py:21]: Confirmed working with unit test.

File: Debug/Proxy_DEBUG_RX.py [hunk 7]
```diff
@@ -1,12 +1,16 @@
...
 
-sys.path.append(os.path.join(os.path.dirname(__file__), "../Faraday_Proxy_Tools/"))  #Append path to common tutorial FaradayIO module
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
 #Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
```
Comment (kb1lqd) [inline Debug/Proxy_DEBUG_RX.py:13]: Confirmed working.

File: Device_Test/devicetest.py [hunk 8]
```diff
@@ -3,13 +3,15 @@
...
 
 #Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import telemetryparser
-from FaradayIO import gpioallocations
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import telemetryparser
+from faraday.proxyio import gpioallocations
```
Comment (kb1lqd) [inline Device_Test/devicetest.py:14]: All device tests seem to work.

File: Tutorials/Python_Developer_Tutorials/foundation/Commanding-Local/Proxy-Local-Commands-Tutorial.py [hunk 9]
```diff
@@ -1,20 +1,21 @@
...
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../../..'))
 
 # Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import gpioallocations
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import gpioallocations
```
Comment (kb1lqd) [inline Tutorials/Python_Developer_Tutorials/foundation/Commanding-Local/Proxy-Local-Commands-Tutorial.py:18]: Works as expected.

File: Tutorials/Python_Developer_Tutorials/foundation/Commanding-Remote-RF/Tutorial_Remote_Command.py [hunk 10]
```diff
@@ -1,15 +1,18 @@
...
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../../..'))
 
 #Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
-from FaradayIO import gpioallocations
-import time
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import gpioallocations
```
Comment (kb1lqd) [inline Tutorials/Python_Developer_Tutorials/foundation/Commanding-Remote-RF/Tutorial_Remote_Command.py:15]: Works!

File: Tutorials/Python_Developer_Tutorials/foundation/RF-Transmit-Receive-Packet/Tutorial_Exp_RF_Packet_TX-User-Input.py [hunk 11]
```diff
@@ -1,13 +1,16 @@
...
-sys.path.append(os.path.join(os.path.dirname(__file__), "../../../../Faraday_Proxy_Tools"))  #Append path to common tutorial FaradayIO module
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../../..'))
 
-#Imports - Faraday Specific
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
+# Imports - Faraday Specific
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
```
Comment (kb1lqd) [inline Tutorials/Python_Developer_Tutorials/foundation/RF-Transmit-Receive-Packet/Tutorial_Exp_RF_Packet_TX-User-Input.py:13]: Good but I see error unrelated from a prior update that causes a type error: 
image

Comment (kb1lqd) [reply Tutorials/Python_Developer_Tutorials/foundation/RF-Transmit-Receive-Packet/Tutorial_Exp_RF_Packet_TX-User-Input.py:13]: I created https://github.com/FaradayRF/Faraday-Software/issues/158 to handle this.

File: Tutorials/Python_Developer_Tutorials/foundation/Telemetry-Parsing/Tutorial_Telemetry_Parsing.py [hunk 12]
```diff
@@ -1,14 +1,18 @@
...
-from FaradayIO import telemetryparser
-from FaradayIO import cc430radioconfig
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../../..'))
+
+# Imports - Faraday Specific
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
+from faraday.proxyio import telemetryparser
+from faraday.proxyio import cc430radioconfig
```
Comment (kb1lqd) [inline Tutorials/Python_Developer_Tutorials/foundation/Telemetry-Parsing/Tutorial_Telemetry_Parsing.py:15]: Works!

File: Tutorials/start/hello-world.md [hunk 13]
```diff
@@ -21,13 +21,18 @@ This subpackage abstracts commands one might wish to send to Faraday. IT contain
...
-sys.path.append(os.path.join(os.path.dirname(__file__), "../../Faraday_Proxy_Tools")) #Append path to common tutorial FaradayIO module
 
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
+
+# Imports - Faraday Specific
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
```
Comment (kb1lqd) [inline Tutorials/start/hello-world.md:35]: Looks like correctly updated code block.

File: Tutorials/start/hello-world.py [hunk 14]
```diff
@@ -1,10 +1,15 @@
...
-sys.path.append(os.path.join(os.path.dirname(__file__), "../../Faraday_Proxy_Tools"))  #Append path to common tutorial FaradayIO module
 
-from FaradayIO import faradaybasicproxyio
-from FaradayIO import faradaycommands
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
+
+# Imports - Faraday Specific
+from faraday.proxyio import faradaybasicproxyio
+from faraday.proxyio import faradaycommands
```
Comment (kb1lqd) [inline Tutorials/start/hello-world.py:12]: Works!

**Review by kb1lqd — APPROVED** [review]
All seems to work through unit testing and anything not tested looks like a valid update/syntax. Passes TravisCI.

**Pull request merged**